### PR TITLE
Add linter config for Python and CI integration on PRs

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -359,6 +360,19 @@ python3 DeepSpeech.py --n_hidden 2048 --checkpoint_dir path/to/checkpoint/folder
...
 
+## Contribution guidelines
+
+This repository is governed by Mozilla's code of conduct and etiquette guidelines. For more details, please read the [Mozilla Community Participation Guidelines](https://www.mozilla.org/about/governance/policies/participation/).
+
+Before making a Pull Request, check your changes for basic mistakes and style problems by using a linter. We have cardboardlinter setup in this repository, so for example, if you've made some changes and would like to run the linter on just the changed code, you can use the follow command:
+
+```bash
+pip install pylint cardboardlint
+cardboardlinter --refspec master
```
Comment (lissyx) [inline README.md]: Can we have doc on setup local pre-commit hook ?

File: util/config.py [hunk 2]
```diff
@@ -45,15 +46,15 @@ def initialize_globals():
         FLAGS.dropout_rate6 = FLAGS.dropout_rate
 
     # Set default checkpoint dir
-    if len(FLAGS.checkpoint_dir) == 0:
-        FLAGS.checkpoint_dir = xdg.save_data_path(os.path.join('deepspeech','checkpoints'))
+    if not FLAGS.checkpoint_dir == 0:
```
[the remark this answers is not in the document]
Comment (reuben) [reply util/config.py]: Yep, forgot to remove the == 0, nice catch!

File: README.md (hunk 1 above)
Comment (reuben) [reply README.md]: I don't know how to do that in a cross-platform way.

**Comment by reuben** [comment]
@lissyx I added pre-commit hook instructions to the README in f27ed52

**Pull request merged**